Plugins/NativeCamera/Source/NativeCameraAndroid.cpp

```diff
@@ -0,0 +1,411 @@
+            ACameraDevice_createCaptureSession(cameraDevice, outputs, &sessionStateCallbacks, &textureSession);
+
+            // Start capturing continuously
+            ACameraCaptureSession_setRepeatingRequest(textureSession, &captureCallbacks, 1, &request, nullptr);
+
+            glGenTextures(1, &cameraRGBATextureId);
+            glBindTexture(GL_TEXTURE_2D, cameraRGBATextureId);
+            glTexImage2D(GL_TEXTURE_2D, 0, GL_RGBA, this->width, this->height, 0, GL_RGBA, GL_UNSIGNED_BYTE, nullptr);
+            glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_MAG_FILTER, GL_LINEAR);
+            glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_MIN_FILTER, GL_LINEAR);
+            glBindTexture(GL_TEXTURE_2D, 0);
+
+            glGenFramebuffers(1, &clearFrameBufferId);
+            glBindFramebuffer(GL_FRAMEBUFFER, clearFrameBufferId);
+            //auto bindFrameBufferTransaction{ GLTransactions::BindFrameBuffer(clearFrameBufferId) };
+            glFramebufferTexture2D(GL_FRAMEBUFFER, GL_COLOR_ATTACHMENT0, GL_TEXTURE_2D, cameraRGBATextureId, 0);
+
+            glBindFramebuffer(GL_FRAMEBUFFER, 0);
+
+            cameraShaderProgramId = CreateShaderProgram(CAMERA_VERT_SHADER, CAMERA_FRAG_SHADER);
+        });
+    }
+
+    CameraInterfaceAndroid::~CameraInterfaceAndroid()
+    {
+        // Stop recording to SurfaceTexture and do some cleanup
+        ACameraCaptureSession_stopRepeating(textureSession);
+        ACameraCaptureSession_close(textureSession);
+        ACaptureSessionOutputContainer_free(outputs);
+        ACaptureSessionOutput_free(output);
+
+        ACameraDevice_close(cameraDevice);
+        ACameraManager_delete(cameraManager);
+
+        // Capture request for SurfaceTexture
+        ANativeWindow_release(textureWindow);
+        ACaptureRequest_free(request);
+    }
+
+    void CameraInterfaceAndroid::UpdateCameraTexture(bgfx::TextureHandle textureHandle)
+    {
+        if (cameraRGBATextureId) {
+            surfaceTexture.updateTexture();
+
+            glBindFramebuffer(GL_FRAMEBUFFER, clearFrameBufferId);
+            glViewport(0, 0, width, height);
+            glUseProgram(cameraShaderProgramId);
+
+            // Configure the camera texture
+            auto cameraTextureUniformLocation{ glGetUniformLocation(cameraShaderProgramId, "cameraTexture") };
+            glUniform1i(cameraTextureUniformLocation, GetTextureUnit(GL_TEXTURE0));
+            glActiveTexture(GL_TEXTURE0);
+            glBindTexture(GL_TEXTURE_EXTERNAL_OES, cameraOESTextureId);
+            glTexParameteri(GL_TEXTURE_EXTERNAL_OES, GL_TEXTURE_MIN_FILTER, GL_NEAREST);
+            glTexParameteri(GL_TEXTURE_EXTERNAL_OES, GL_TEXTURE_MAG_FILTER, GL_NEAREST);
+            glBindSampler(0, 0);
+
+            // Draw the quad
+            glDrawArrays(GL_TRIANGLE_STRIP, 0, 3);
+
+            glUseProgram(0);
+            glBindFramebuffer(GL_FRAMEBUFFER, 0);
+
+            bgfx::overrideInternal(textureHandle, cameraRGBATextureId);
```

Comment:
As on the iOS one, is this going to encounter problems if we're not on the graphics thread?

Reply:
Same comment as above. Wait to test with new threading model.